### DISCUSSION TITLE
Netherlands: All companies (the right way)

### DISCUSSION THREAD

**Discussion by DannyOosterveer** [JaccoR/hass-entso-e#92] [Price Modifyer Templates]
Changed the calculation, and updated with 2023's energy, VAT and updated tariffs:

{% set s = {
  "All-in-power": 0.025,
  "ANWB_Energie": 0.0175,
  "easyEnergy": 0.0180,
  "Eneco": 0.02059,
  "EnergyZero": 0.0175,
  "Frank_Energie": 0.0210,
  "GroeneStroomLokaal": 0.0175,
  "Mijndomein_Energie": 0.0175,
  "NextEnergy": 0.0183,
  "Tibber": 0.0020,
  "Vrijopnaam": 0.026,
  "ZonderGas": 0.0175,
  "Zonneplan": 0.0025,

  "energie_belasting": 0.12599,
  "VAT": 1.21
} %}

{{ ((current_price + s.Zonneplan + s.energie_belasting) * s.VAT) | float}}

**Comment by hocomadvies** [comment]
Since I have my Energie from Frank_Energie, I adapted the part `(current_price + s.Zonneplan)` to `(current_price + s.Frank_Energie)` but pasting the above complete code result in an error. Or must I edit the sample code from the install-node including this part? How to get this working :0 Thanks.

**Reply by DannyOosterveer** [reply]
Alternatively use the code below, and replace the value of extra_cost with the value from your energy provider from the list above:

{% set s = {"extra_cost": 0.002,"energie_belasting": 0.04010,"ode_belasting": 0.03325,"VAT": 1.09} %} {{(((current_price + s.extra_cost) * s.VAT) + s.energie_belasting + s.ode_belasting) | float}}

**Reply by PjotrBaas** [reply]
>..... How to get this working :0 Thanks.

It results in error because: count the ")", there's one to much.

**Comment by woytekbode** [comment]
Actually, VAT is calculated for energiebelasting and ODE as well. However, some website show the numbers including VAT. Therefore, it's important to either use the incl. VAT prices and not calculate the VAT or use the excl. VAT prices and calculate the VAT yourself.

**Comment by fouries** [comment]
这是来自我的邮箱的自动回复邮件。你好，已收到您的邮件，我会尽快给你回复。

**Comment by ardonjr** [comment]
VAT mag vanaf 1/1/2023 weer naar 21% 
Daarbij is GroeneStroomLokaal onderdeel van EnergyZero, deze hanteren dus ook beiden dezelfde inkoopkosten. Welke op moment van schrijven zijn gewijgd naar 0,02118

**Reply by DannyOosterveer** [reply]
Dank! Meteen ook aangepast op https://doe-duurzaam.nl/dossier/dynamische-energietarieven/.

**Comment by fouries** [comment]
这是来自我的邮箱的自动回复邮件。你好，已收到您的邮件，我会尽快给你回复。

**Comment by joostbijl** [comment]
The jinja in the OP fails because of an extra ", a comma instead of period and a missing bracket. The code should be:

```
{% set s = {
  "All-in-power": 0.025,
  "ANWB_Energie": 0.0175,
  "easyEnergy": 0.0180,
  "EnergyZero": 0.0175,
  "Frank_Energie": 0.0210,
  "GroeneStroomLokaal": 0.0175,
  "Mijndomein_Energie": 0.0175,
  "NextEnergy": 0.0183,
  "Tibber": 0.0020,
  "Vrijopnaam": 0.026,
  "ZonderGas": 0.0175,
  "Zonneplan": 0.0025,

  "energie_belasting": 0.12599,
  "VAT": 1.21
} %}

{{ ((current_price + s.Zonneplan + s.energie_belasting) * s.VAT) | float}}
```

**Reply by schellekensonline** [reply]
Since prices can become negative shouldn't it be calculated as : ? 

    {% set s = { "NextEnergy": 0.0183, "energie_belasting": 0.12599, "VAT": 0.21 } %}
    {{( current_price + (current_price|abs) * s.VAT + (s.NextEnergy + s.energie_belasting) * (1+s.VAT) ) | float}}

**No VAT is payed when prices are negative:-D** So no |abs, the following it is, as mentioned in earlier posts 

    {{( (current_price + s.NextEnergy + s.energie_belasting) * (1+s.VAT) ) | float}}

**Reply by DannyOosterveer** [reply]
> The jinja in the OP fails because of an extra ", a comma instead of period and a missing bracket. The code should be:
> 
> ```
> {% set s = {
>   "All-in-power": 0.025,
>   "ANWB_Energie": 0.0175,
>   "easyEnergy": 0.0180,
>   "EnergyZero": 0.0175,
>   "Frank_Energie": 0.0210,
>   "GroeneStroomLokaal": 0.0175,
>   "Mijndomein_Energie": 0.0175,
>   "NextEnergy": 0.0183,
>   "Tibber": 0.0020,
>   "Vrijopnaam": 0.026,
>   "ZonderGas": 0.0175,
>   "Zonneplan": 0.0025,
> 
>   "energie_belasting": 0.12599,
>   "VAT": 1.21
> } %}
> 
> {{ ((current_price + s.Zonneplan + s.energie_belasting) * s.VAT) | float}}
> ```

Fixed, the typo, thnx!

**Comment by fouries** [comment]
这是来自我的邮箱的自动回复邮件。你好，已收到您的邮件，我会尽快给你回复。

**Comment by groetg** [comment]
Ik heb ANWB Energie en ik ervaar iets geks. 

Current hour volgens de ANWB app is nu 0.08 cent/kwh,

Ik maak dus het volgende sommetje:
current hour: 0.08 cent/kwh
ANWB_Energie: 0.0175 cent/kwh
energie_belasting: 0.12599 cent/kwh
Totaal excl. BTW: 0.08 + 0.0175 + 0.12599 = 0.22349 cent/kwh
Totaal incl. BTW: 0.22349 x 1.21 = 0,2704229 cent/kwh

Terwijl ik bij Current electricity market price de volgende prijs zie: 0,15815 €/kWh

Mijn template: 
{% set s = {"ANWB_Energie": 0.0175,"energie_belasting": 0.12599,"VAT": 1.21} %} {{ ((current_price+s.ANWB_Energie+s.energie_belasting) * s.VAT) | float}}

**Comment by PV-Web** [comment]
How and where can I find the new values for 2024? Im using NextEnergy but would love to learn how to find them myself. It is a bit confusing at the moment.

**Reply by DannyOosterveer** [reply]
You can find the most accurate dynamic hourly rates here: https://doe-duurzaam.nl/dossier/dynamische-energietarieven/

**Comment by tjallingk** [comment]
2024 de belasting is nu lager ook vroeg ik me af kunnen we rekening houden met de hogere tarief als je meer dan 10k gebruikt? bron: https://support.tibber.com/nl/articles/4406583-kosten-bij-tibber

**Reply by DannyOosterveer** [reply]
Klopt, dus ipv 0,12599 kun je nu 0,1088 hanteren. Als je meer dan 10.000 (netto) gebruikt, dan is het tarief 0,09037. (https://www.belastingdienst.nl/wps/wcm/connect/bldcontentnl/belastingdienst/zakelijk/overige_belastingen/belastingen_op_milieugrondslag/tarieven_milieubelastingen/tabellen_tarieven_milieubelastingen?projectid=6750bae7-383b-4c97-bc7a-802790bd1110)

**Comment by casperbuys** [comment]
Ik gebruik de volgende multiplier bij Eneco en ik heb tot dusver altijd de goede prijs zichtbaar:

{{ ((current_price + 0.02059 + 0.10522546) * 1.21) | float}}

**Comment by Hein1965** [comment]
Weet iemand waar ik de extra_cost voor de energieleverancier Essent kan vinden (in het kader van ENTSO-e)?
Overal waar ik kijk op internet staan lijstjes met leveranciers en hun extra_cost, maar Essent zie ik er nooit tussen staan.
Jullie hulp wordt gewaardeerd..!

Edit 5 Oktober: 
Pfft... Essent heeft helemaal geen dynamische prijzen, dan kun je zoeken tot je een ons weegt natuurlijk...

**Reply by DannyOosterveer** [reply]
Bij Essent betaal je 0,02089 euro per kWh (excl. btw) (bron https://doe-duurzaam.nl/dossier/dynamische-energietarieven/)

**Comment by oliv3r** [comment]
Why are we calculating the VAT via the price modifier template, and not just use the plugin's 'vat tariff'? I would expect that the add-on would also just do 'price-modifier * 1+VAT tariff anyway?

**Comment by bmmaddy** [comment]
anyone has template for next energy ?

**Reply by jaspersweere** [reply]
NextEnergy prices: I have done it like so, but i'm off by 2 cents... HELP!

{% set inkoop = 0.0219 %}
{% set belasting = 0.11084 %}
{% set btw = 1.21 %}
{{ ((current_price + inkoop) * btw + (belasting * btw)) | round(3) }}
